### PR TITLE
Icon titles

### LINKED ISSUE #77: Need to make sure SVGs have titles
@robdodson commented on [Thu Dec 03 2020](https://github.com/GoogleChrome/dcc-private/issues/401)

We use a lot of SVGs for icons. We should make sure they have accessible titles.

### PR DESCRIPTION
**Pull request by samthor**
Fixes #77

### CODE REVIEW

File: site/_includes/macros/navigation-tree-sections.njk [hunk 1]
```diff
@@ -34,7 +34,7 @@ and marks it as active.
       {% set postTitle = item.title | i18n(locale) %}
       <a href="{{ item.url }}" class="navigation-tree__link" target="_blank" rel="noopener">
         <span class="display-inline-flex align-center">
-          {{ postTitle }} {{ icon('external-link', {className: 'width-300 gap-left-100'}) }}
+          {{ postTitle }} {{ icon('external-link', {label: 'external link', className: 'width-300 gap-left-100'}) }}
```
Comment (dotproto) [inline site/_includes/macros/navigation-tree-sections.njk:37]: Should this label be localized?

File: site/_includes/macros/navigation-tree-sections.njk [hunk 2]
```diff
@@ -16,7 +16,7 @@ and marks it as active.
   {% if item.sections %}
     <button class="navigation-tree__link button" data-expandable aria-expanded="{{ item.active }}">
       <div class="navigation-tree__icon">
-        {{ icon('arrow-right', {label: 'toggle'}) }}
+        {{ icon('arrow-right', {label: 'toggle', hidden: true}) }}
```
Comment (dotproto) [inline site/_includes/macros/navigation-tree-sections.njk:19]: Should this label be localized?

Comment (dotproto) [inline site/_includes/macros/navigation-tree-sections.njk:19]: I wasn't quite sure why this change was added, but after [checking MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Techniques/Using_the_aria-hidden_attribute) and viewing a page on the preview site it appears that this change hides non-meaningful content from the accessibility tree. Nice :)

File: site/_includes/partials/footer.njk [hunk 3]
```diff
@@ -90,7 +90,7 @@
...
...6.773.58 1.667.58 2.68 0 .12-.007.24-.02.36a3.41 3.41 0 00-.02.2h-7.64c.066.907.393 1.607.98 2.1a3.02 3.02 0 002 .74c1.146 0 2.026-.533 2.64-1.6l1.88.9a5.2 5.2 0 01-1.84 1.96c-.774.48-1.687.72-2.74.72zm2.52-6.5a2.433 2.433 0 00-.32-.98c-.187-.333-.48-.613-.88-.84-.387-.24-.874-.36-1.46-.36-.654 0-1.22.2-1.7.6-.467.4-.787.927-.96 1.58h5.32zm4.45-3.74h2.08v1.48h.12c.214-.507.6-.927 1.16-1.26.56-.347 1.134-.52 1.72-.52.48 0 .874.067 1.18.2v2.3a3.35 3.35 0 00-1.52-.34c-.72 0-1.326.28-1.82.84-.48.547-.72 1.22-.72 2.02V15h-2.2V5.08zm11.441 10.24c-1.134 0-2.074-.26-2.82-.78-.734-.533-1.254-1.2-1.56-2l1.96-.84c.226.547.553.96.98 1.24.44.28.92.42 1.44.42.546 0 .993-.1 1.34-.3.346-.213.52-.487.52-.82 0-.32-.14-.58-.42-.78-.28-.213-.74-.4-1.38-.56l-1.36-.32a4.5 4.5 0 01-1.94-.96c-.534-.48-.8-1.1-.8-1.86 0-.92.38-1.647 1.14-2.18.76-.547 1.693-.82 2.8-.82.92 0 1.733.193 2.44.58.72.387 1.24.947 1.56 1.68l-1.94.82a1.87 1.87 0 00-.86-.92c-.387-.2-.807-.3-1.26-.3-.44 0-.827.1-1.16.3-.334.187-.5.44-.5.76 0 .28.113.507.34.68.24.173.613.32 1.12.44l1.48.36c1 .253 1.746.64 2.24 1.16.493.507.74 1.12.74 1.84 0 .587-.18 1.12-.54 1.6-.347.48-.834.86-1.46 1.14-.614.28-1.314.42-2.1.42z" fill="#5F6368"/></svg>
```
Comment (dotproto) [inline site/_includes/partials/footer.njk:93]: Should this aria-label be localized?

File: site/_includes/partials/navigation-rail.njk [hunk 4]
```diff
@@ -3,10 +3,10 @@
 <navigation-rail role="navigation" class="lg:pad-left-200 lg:pad-right-200" aria-label="{{ 'i18n.common.primary' | i18n(locale) }}" tabindex="-1">
   <div class="display-flex lg:display-none align-center pad-200 hairline-bottom">
     <button class="navigation-rail__close button display-flex align-center justify-content-center width-700 height-700" aria-label="close">
-      {{ icon('close') }}
+      {{ icon('close', {label: 'close'}) }}
```
Comment (dotproto) [inline site/_includes/partials/navigation-rail.njk:6]: Should this label be localized? The button already has an appropriate aria-label, should this icon be hidden from the accessibility tree?

File: site/_includes/partials/navigation-rail.njk [hunk 5]
```diff
@@ -3,10 +3,10 @@
...
     </button>
     <div class="display-flex pad-left-200 pad-right-200" class="navigation-rail__head">
-      {{ icon('logo') }}
+      {{ icon('logo', {label: 'Google Chrome logo'}) }}
```
Comment (dotproto) [inline site/_includes/partials/navigation-rail.njk:9]: Should this label be localized?

File: site/_includes/partials/cookie-banner.njk [hunk 6]
```diff
@@ -8,7 +8,7 @@
   <div class="cookie-banner__controls">
     <a class="material-button button-text color-blue-medium display-inline-flex align-center"
        href="{{ site.cookiesCtaUrl }}">
-      More details {{ icon('external-link', {className: 'gap-left-200'}) }}
+      More details {{ icon('external-link', {label: 'external link', className: 'gap-left-200'}) }}
```
Comment (dotproto) [inline site/_includes/partials/cookie-banner.njk:11]: Should this label be localized?

**Review by dotproto — APPROVED** [review]
LGTM. A couple requests for clarification, but nothing major or blocking.

File: site/_includes/macros/cards/featured-docs-card.njk [hunk 7]
```diff
@@ -8,10 +8,6 @@
     <span class="gap-left-300 flex-1 user-select-none">
       {{ 'i18n.nav.side_nav.docs' | i18n(locale) }}
     </span>
-    {# <div class="card-title-bar__controls">
-      {{ icon('keyboard-arrow-left') }}
-      {{ icon('keyboard-arrow-right') }}
-    </div> #}
```
[the remark this answers is not in the document]
Comment (samthor) [reply site/_includes/macros/cards/featured-docs-card.njk:14]: Good call.

File: site/_includes/macros/navigation-tree-sections.njk (hunk 1 above)
Comment (samthor) [reply site/_includes/macros/navigation-tree-sections.njk]: Done

File: site/_includes/macros/navigation-tree-sections.njk (hunk 2 above)
Comment (samthor) [reply site/_includes/macros/navigation-tree-sections.njk]: I've hidden this one and used the button outside it instead.

File: site/_includes/macros/navigation-tree-sections.njk (hunk 1 above)
Comment (samthor) [reply site/_includes/macros/navigation-tree-sections.njk]: Done

File: site/_includes/partials/footer.njk (hunk 3 above)
Comment (samthor) [reply site/_includes/partials/footer.njk:93]: No, probably not since it's a brand and the SVG image literally has the words in it.

File: site/_includes/macros/navigation-tree-sections.njk (hunk 2 above)
Comment (samthor) [reply site/_includes/macros/navigation-tree-sections.njk]: Yeah. I've removed the label too, since it's pointless if it's hidden.

File: site/_includes/partials/navigation-rail.njk (hunk 4 above)
Comment (samthor) [reply site/_includes/partials/navigation-rail.njk]: No, good call. I've hidden it.

Comment (samthor) [reply site/_includes/partials/navigation-rail.njk]: No, good call. I've hidden it.

File: site/_includes/partials/cookie-banner.njk (hunk 6 above)
Comment (samthor) [reply site/_includes/partials/cookie-banner.njk]: Done

File: site/_includes/partials/navigation-rail.njk (hunk 5 above)
Comment (samthor) [reply site/_includes/partials/navigation-rail.njk:9]: Yes, but, I've removed "logo", so it won't need it (since we don't translate the browser name).
